CMSIS/Core/Include/core_armv81mml.h

```diff
@@ -97,6 +97,29 @@
     #define __DSP_USED         0U
   #endif
 
+#elif defined (__ti__)
```

Comment:
Adding this for every compiler seems not to be maintainable on the long run when adding more compilers. We need to look for another solution. Can we move this somehow into the compiler header? Or reduce to be compiler independent by adding a define-mapping to compiler headers?

Reply:
In general, I agree -- there is a lot of duplication here, and there are also some target specific dependencies in a handful of the files (depending on whether the target cares about the compiler enabling FPU instructions, DSP instructions, or not).  Also, tiarmclang also defines __GNUC__, which ties it together.  I'm willing to do what I can for tiarmclang, but I think someone needs to look at the design holistically and then refactor in a way that makes sense.  I think that's beyond the scope of this pull request, although I absolutely understand why you'd call it out.

Reply:
Yes, agree. We should not do to much at once. I think we eventually can move the defines into the compiler headers. Let's leave it as is for now. I'll put me a note to address this change in CMSIS 6.